engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -253,7 +266,7 @@ namespace dmGameObject
 
     Collection* AllocCollection(const char* name, HRegister regist, uint32_t max_instances)
     {
-        Collection* collection = new Collection(0, 0, max_instances);
+        Collection* collection = new Collection(0, 0, max_instances, GetInputStackDefaultCapacity(regist));
```

Comment:
I wonder why the "max_input_stack_entries" doesn't take the same code path as the other configurable "max_instances"?

Reply:
I guess I could make it the same, it's just that that max_instances gets passed around through 3 functions before reaching that constructor and the indirection felt unnecessary in this case. I don't really know why is there so much indirection or what's the better approach here.

Reply:
I can't say I like passing those things around either.
Also, in the future, this library should take a create params struct, like our other systems do. E.g. to dmGameObject::NewRegister(create_params). In order to reduce the amount of overhead for passing/storing/reading the settings.
But, that's for the future. Let's keep this for now.